rwengine/src/audio/SoundManager.cpp

```diff
@@ -166,10 +185,82 @@ void SoundManager::SoundSource::loadFromFile(const std::string& filename) {
         }
         av_free_packet(&readingPacket);
     }
+#else
+
+    AVFrame* resampled = nullptr;
+
+    while (av_read_frame(formatContext, &readingPacket) == 0) {
+        if (readingPacket.stream_index == audioStream->index) {
+
+            int sendPacket = avcodec_send_packet(codecContext, &readingPacket);
+            av_packet_unref(&readingPacket);
+            int receiveFrame = 0;
+
+
+            while ((receiveFrame = avcodec_receive_frame(codecContext, frame)) == 0) {
+                if(!swr) {
+                    if(frame->channels == 1 || frame->channel_layout == 0) {
+                        frame->channel_layout = av_get_default_channel_layout(1);
+                    }
+                    swr = swr_alloc_set_opts(nullptr,
+                                             AV_CH_LAYOUT_STEREO,                           // output channel layout
+                                             kOutputFMT,                                    // output format
+                                             frame->sample_rate,                            // output sample rate
+                                             frame->channel_layout,                         // input channel layout
+                                             static_cast<AVSampleFormat>(frame->format),    // input format
+                                             frame->sample_rate,                            // input sample rate
+                                             0,
+                                             nullptr);
+                    if (!swr) {
+                        RW_ERROR("Resampler has not been successfully allocated.");
+                        return;
+                    }
+                    swr_init(swr);
+                    if (!swr_is_initialized(swr)) {
+                        RW_ERROR("Resampler has not been properly initialized.");
+                        return;
+                    }
+                }
+                // Decode audio packet
+
```

Comment:
I assume this empty line was supposed to be above the comment, not after it.

Reply:
Reformatted it a bit. Should now be a little better. :)